File: src/003_moveit_config/scripts/mycode/tag_detection.py
import rospy
from std_msgs.msg import String
from apriltag_ros.msg import AprilTagDetectionArray

def Localizer_init():
    
    sub = rospy.Subscriber("tag_detections", AprilTagDetectionArray, number_callback, queue_size=10)
    
def number_callback(msg):

    last_msg_ = msg
    print(msg)
    

if __name__ == "__main__":

    rospy.init_node("apriltag_detector_subscriber")
    localizer = Localizer_init()
    rospy.spin()

                                                                                                                            

File: src/003_moveit_config/src/tag_detection.cpp
// #include "ros/ros.h"
// #include "apriltag_ros/AprilTagDetectionArray.h"
// #include <iostream>
// class Localizer
// {
// public:
//   Localizer(ros::NodeHandle& nh)
//   {
//       ar_sub_ = nh.subscribe<apriltag_ros::AprilTagDetectionArray>("tag_detections", 1, &Localizer::number_callback, this);
//   }
//   void number_callback(const apriltag_ros::AprilTagDetectionArray::ConstPtr& msg)
//   {
//       last_msg_ = msg;
//       ROS_INFO_STREAM(last_msg_->detections[0].pose.pose.pose);
//       //ROS_INFO("the number %d ",);
//       //ROS_INFO("%s\n", s.data.c_str());
//   }

//   ros::Subscriber ar_sub_;
//   apriltag_ros::AprilTagDetectionArrayConstPtr last_msg_;
// };


// int main(int argc, char **argv)
// {
//     ros::init(argc, argv,"apriltag_detector_subscriber");
//     ros::NodeHandle node_obj;
//     Localizer localizer(node_obj);
//     //ros::Subscriber number_subscriber = node_obj.subscribe("tag_detections",1,number_callback);
//     ROS_INFO("1Vision node starting");
//     ros::spin();
//     //return 0;
// }
